runtime/onert/core/src/backend/basic/Tensor.cc

```diff
@@ -32,40 +32,23 @@ void Tensor::setShape(const ir::Shape &new_shape) { _info.shape(new_shape); }
 
 bool Tensor::applyShape(const ir::Shape &new_shape)
 {
-  bool previously_dynamic = is_dynamic();
+  if (_buffer != nullptr && new_shape == _info.shape())
+    return true;
 
-  auto allocTensorMem = [&]() {
-    auto capacity = total_size();
-    assert(_dynamic_mem_mgr);
-    auto alloc = _dynamic_mem_mgr->allocate(this, capacity);
-    setBuffer(alloc);
-  };
-
-  if (!previously_dynamic || buffer() == nullptr)
+  // Always set shape - when buffer with same or larger size was already allocated, shape could
+  // differ
```

Comment:
(Optional) It may make sense to use `vary` than `differ`
```suggestion
  // vary
```

Reply:
I just found out the comment already exists.